tools/autograd/gen_variable_type.py

```diff
@@ -15,106 +16,96 @@
 #       - The function's output is not differentiable
 #       - The function has no data dependency on its input
 #
-#     These are currently called "fallthrough" functions, although
-#     they are not entirely fallthrough; for example, if the function
-#     in question returns a tensor, we have to wrap it in a
-#     (does not require grad) variable to abide by the API contract
-#     of Variable.
-#
 #   - Some function don't need a backwards implementation because they
-#     are implement as a composition of other (differentiable) ATen
+#     are implemented as a composition of other (differentiable) ATen
 #     functions.  These are dispatched directly to the Type superclass,
 #     which will in turn dispatch back to VariableType for its
 #     differentiable subcomponents.
+#
+from __future__ import print_function
+import sys
+from .utils import CodeTemplate, nested_dict
+from .gen_autograd import VIEW_FUNCTIONS, template_path, write
+from .gen_autograd_functions import uses_grad
+
+VARIABLE_TYPE_H = CodeTemplate.from_file(template_path + '/VariableType.h')
+VARIABLE_TYPE_CPP = CodeTemplate.from_file(template_path + '/VariableType.cpp')
+
+# These functions are written manually in templates/VariableType.cpp
+MANUAL_IMPLEMENTATIONS = {
+    'contiguous', 'resize_', 'resize_as_'
+}
+
+# These functions we don't want to record for tracing, because we always want
+# to trace their constituent parts.  This is a temporary hack in lieue
+# of proper scopes, where subsequent compilation passes can ask for the unfolding
+# on demand.  Only concrete ATen methods can be disabled this way; it will have
+# NO EFFECT otherwise.
+DONT_RECORD_TRACE = {
+    'convolution', 'conv1d', 'conv2d', 'conv3d', 'conv_transpose1d',
+    'conv_transpose2d', 'conv_transpose3d',
+}
 
-import argparse
-import copy
-import os
-import re
-import yaml
-from collections import defaultdict
-from .utils import CodeTemplate, write, nested_dict, YamlLoader
+# These functions are not worth profiling because they are very cheap and may
+# be called very often.
+DONT_PROFILE = {
+    'data_ptr', 'get_device', 'is_contiguous', 'is_cuda', 'is_distributed',
+    'is_same_size', 'is_set_to', 'is_signed', 'is_sparse', 'numel',
+    'size', 'storage_offset', 'stride',
+}
 
+DONT_REQUIRE_DERIVATIVE = {
```

Comment:
can you add a comment saying what happens to those functions (do we still generate them? is their backward just an error? are they guaranteed to return things that don't require grad?)?